sdk/clientcore/core/src/main/java/io/clientcore/core/serialization/json/JsonSerializer.java

```diff
@@ -89,6 +86,41 @@ public <T> T deserializeFromBytes(byte[] bytes, Type type) throws IOException {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> T deserializeListOfBinaryData(JsonReader jsonReader) throws IOException {
+        JsonToken token = jsonReader.currentToken();
```

Comment:
Any reason we don't use `readArray` here? This and the `JsonSerializable` implementation are the same except that looks for the `fromJson` method to call and this just uses `BinaryData.fromObject(arrayReader.readerUntyped())`.